### PR TITLE
Fix guild's IconUrl

### PR DESCRIPTION
**Pull request by Scordium**
# Summary
Fixes an issue with guild's icon URL format always being .jpg even if the icon is animated.
Adds a method to construct guild icon URL.

# Changes proposed

- Changed default icon format from jpg to png

-  Added a check in IconUrl property for animated icons.

-  Created a method for constructing icon URL with specified image format and size.

### CODE REVIEW

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 1]
```diff
@@ -503,7 +504,43 @@ internal DiscordGuild()
...
+        /// <returns>The URL of the guild's icon.</returns>
+        public string GetIconUrl(ImageFormat imageFormat, ushort imageSize = 1024)
+        {
+
+            if (imageFormat == ImageFormat.Unknown)
+                throw new ArgumentException("You must specify valid image format.", nameof(imageFormat));
+
+            // Makes sure the image size is in between Discord's allowed range.
+            if (imageSize < 16 || imageSize > 4096)
+                throw new ArgumentOutOfRangeException("Image Size is not in between 16 and 4096: " + nameof(imageSize));
```
Comment (N0D4N) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:522]: I'm not sure about coding standards in D#+, but won't it be better to utilize [this constructor](https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception.-ctor?view=net-6.0#system-argumentoutofrangeexception-ctor(system-string-system-object-system-string)) which will set name of param, and its actual value.
```suggestion
            if (imageSize < 16 || imageSize > 4096)
                throw new ArgumentOutOfRangeException(nameof(imageSize), imageSize, "Image Size is not in between 16 and 4096.");
```

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 2]
```diff
@@ -503,7 +504,43 @@ internal DiscordGuild()
...
+
+            // Checks to see if the image size is not a power of two.
+            if (!(imageSize is not 0 && (imageSize & (imageSize - 1)) is 0))
+                throw new ArgumentOutOfRangeException("Image size is not a power of two: " + nameof(imageSize));
```
Comment (N0D4N) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:526]: Same as above.
```suggestion
                throw new ArgumentOutOfRangeException(nameof(imageSize), imageSize, "Image size is not a power of two.");
```

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 3]
```diff
@@ -63,7 +64,7 @@ public class DiscordGuild : SnowflakeObject, IEquatable<DiscordGuild>
         /// </summary>
         [JsonIgnore]
         public string IconUrl
-            => !string.IsNullOrWhiteSpace(this.IconHash) ? $"https://cdn.discordapp.com/icons/{this.Id.ToString(CultureInfo.InvariantCulture)}/{this.IconHash}.jpg" : null;
+            => !string.IsNullOrWhiteSpace(this.IconHash) ? $"https://cdn.discordapp.com/icons/{this.Id.ToString(CultureInfo.InvariantCulture)}/{this.IconHash}.{(this.IconHash.StartsWith("a_") ? "gif" : "png")}?size=1024" : null;
```
Comment (N0D4N) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:67]: Don't duplicate code, call newly created method instead.
```suggestion
            => this.GetIconUrl(ImageFormat.Auto, 1024);
```

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 4]
```diff
@@ -503,7 +504,43 @@ internal DiscordGuild()
...
+        
+        /// <summary>
+        /// Gets guild's icon URL, in requested format and size.
+        /// </summary>
+        /// <param name="imageFormat">The image format of the icon to get.</param>
+        /// <param name="imageSize">The maximum size of the icon. Must be a power of two, minimum 16, maximum 4096.</param>
+        /// <returns>The URL of the guild's icon.</returns>
+        public string GetIconUrl(ImageFormat imageFormat, ushort imageSize = 1024)
+        {
+
```
Comment (N0D4N) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:516]: According to current implementation of `IconUrl, `IconHash` property can be null or whitespaced, and in that case `IconUrl` returns `null`. Check for it should be added to this method too.

**Review by N0D4N — APPROVED** [review]

**Comment by OoLunar** [comment]
LGTM

**Pull request merged**